.github/workflows/rustls.yml

```diff
@@ -0,0 +1,34 @@
+name: Rustls
```

Comment:
nit, not required: could we use [GHA's matrix support](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) to cut down on # of jobs?

otherwise, good to merge as-is.